src/wp-includes/functions.php

```diff
@@ -2087,9 +2097,29 @@ function wp_mkdir_p( $target ) {
 			}
 		}
 
+		/**
+		 * Fires after the directory is created and the permissions are set.
+		 *
+		 * @since 6.4.0
+		 *
+		 * @param string $target    Full path to the created directory.
+		 * @param int    $dir_perms Directory permissions.
+		 */
+		do_action( 'after_create_directory', $target, $dir_perms );
+
 		return true;
 	}
 
+	/**
+	 * Fires when directory creation fails
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string    $target     Full path to the directory that failed to create
+	 * @param int       $dir_perms  Directory permissions.
+	 */
```

Comment:
```suggestion
	/**
	 * Fires when directory creation fails.
	 *
	 * @since 6.4.0
	 *
	 * @param string $target    Full path to the directory that failed to create
	 * @param int    $dir_perms Directory permissions.
	 */
```
Just adding a `.` and some alignment fixes.